build-logic/java-api-extractor/src/main/java/org/gradle/internal/tools/api/impl/JavaApiMemberWriter.java

```diff
@@ -57,8 +60,12 @@ public void writeClass(ClassMember classMember, Set<MethodMember> methods, Set<F
         for (String permittedSubclass : classMember.getPermittedSubclasses()) {
             apiMemberAdapter.visitPermittedSubclass(permittedSubclass);
         }
+        InnerClassMember declaringInnerClass = innerClasses.stream()
```

Comment:
🤔 I wonder if we can figure that out from access as here:
https://github.com/gradle/gradle/blob/49c24ac23d9381d646c88e6fd731fb005cef7b3a/build-logic/java-api-extractor/src/main/java/org/gradle/internal/tools/api/impl/ApiMemberSelector.java#L73

Reply:
I tested that, but I always get `33` (`ACC_PUBLIC` and `ACC_SUPER`) for `access` for the `classMember`.

Reply:
:+1: It would be good to do a proper review/refactor on it.